flycheck.el

```diff
@@ -2191,7 +2191,10 @@ current buffer as BUFFER.
 
 Return non-nil if the BUFFER is backed by a file, and not
 modified, or nil otherwise."
-  (and (buffer-file-name buffer) (not (buffer-modified-p buffer))))
+  (let ((file-name (buffer-file-name buffer))
+        (not-buffer-modified (not (buffer-modified-p buffer))))
+    (if (not file-name) nil
+      (and (file-exists-p file-name) not-buffer-modified))))
```

Comment:
Isn't this the same as `(and file-name (file-exists-p file-name) …)`?  Then the whole expression shrinks to

```elisp
(and file-name (file-exists-p file-name) (not (buffer-modified-p buffer)))
```